Remove FAQ documentation content, fix links

stichbury opened the pull request:
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
Part of #1985 to improve the FAQ experience. The plan is to:

* Move marketing FAQs to the website
* Add more technical  FAQ content in the documentation where it's relevant (e.g. we now have a spaceflights FAQ)
* Point to the FAQ content from other locations if it's not optimised for search (e.g. write a blog post, create a new 'skinny' FAQ page that passes readers through to the content)

## Development notes
Removed FAQ content and graphics that are no longer needed
Fixed links
Moved architecture and kedro principles docs into "Resources" for now (but their time is limited)

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

astrojuanlu commented on docs/source/development/commands_reference.md:
````diff
@@ -316,7 +316,7 @@ Call the `run()` method of the `KedroSession` defined in `kedro.framework.sessio
 ```
 
-`KedroContext` can be extended in `run.py` (`src/<package_name>/run.py`). In order to use the extended `KedroContext`, you need to set `context_path` in the [`pyproject.toml` configuration file](../faq/architecture_overview.md#kedro-project).
+`KedroContext` can be extended in `run.py` (`src/<package_name>/run.py`). In order to use the extended `KedroContext`, you need to set `context_path` in the `pyproject.toml` configuration file.
````
Maybe link to https://kedro.readthedocs.io/en/stable/kedro_project_setup/settings.html ?

stichbury replied:
I don't know @astrojuanlu. I'd be happy to add it, but I looked at the page and `context_path` isn't mentioned there because it covers `settings.py`. Perhaps that page itself needs to be extended to cover `pyproject.toml` as well? cc @merelcht

stichbury replied:
Makes sense, and I've created an issue for that very thing.

astrojuanlu replied:
xref gh-2377 👍🏽

astrojuanlu commented on docs/source/get_started/install.md:
````diff
@@ -43,7 +43,9 @@ We suggest you create a new Python virtual environment for *each* new Kedro proj
 
-Depending on your preferred Python installation, you can alternatively create virtual environments to work with Kedro using `venv` or `pipenv` instead of `conda`, as described in the [FAQ](../faq/faq.md)
+```{note}
+Depending on your preferred Python installation, you can alternatively create virtual environments to work with Kedro using `venv` or `pipenv` instead of `conda`.
````
xref https://github.com/kedro-org/kedro/issues/2360#issuecomment-1447971885

Just for the record, do you intend to add such instructions in this PR?

stichbury replied:
I wasn't planning to because #2360 is coming (we should prioritise for next sprint) and we'll have both PRs in the docs before next release, so the absence should be OK.

astrojuanlu commented on docs/source/get_started/install.md:
```diff
@@ -95,8 +97,46 @@ You should see an ASCII art graphic and the Kedro version number: for example,
+
+### How do I upgrade Kedro?
+
+We use [Semantic Versioning](https://semver.org/). The best way to safely upgrade is to check our [release notes](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) for any notable breaking changes. Follow the steps in the migration guide included for that specific release.
```
> We use Semantic Versioning

Do we? 😅 

> major version zero [means] Anything MAY change at any time

(from https://semver.org/#semantic-versioning-specification-semver)

I'd need someone from engineering to weigh in

stichbury replied:
cc @merelcht on this. I will remove that first sentence anyway, since I don't think it adds anything to the docs at this point. Let's add it to the FAQ list though.

astrojuanlu commented on docs/source/get_started/install.md:
````diff
@@ -95,8 +97,46 @@ You should see an ASCII art graphic and the Kedro version number: for example,
+pip install kedro -U
+```
+
+When migrating an existing project to a newer Kedro version, make sure you also update the `project_version` in your `pyproject.toml` file from the project root directory or, for projects generated with Kedro<0.17.0, in your `ProjectContext`, which is found in `src/<package_name>/run.py`.
````
Since this is about older Kedro versions, maybe put it in a note or warning?

astrojuanlu commented on docs/source/get_started/install.md:
````diff
@@ -95,8 +97,46 @@ You should see an ASCII art graphic and the Kedro version number: for example,
+To try out latest, unreleased functionality, run the following installation command:
+
+```console
+pip install git+https://github.com/kedro-org/kedro.git@develop
````
Wondering if we should clarify `main` vs `develop` here 🤔

astrojuanlu commented on docs/source/get_started/kedro_concepts.md:
```diff
@@ -60,7 +60,7 @@ project-dir         # Parent directory of the template
 ├── logs            # Project output logs (not committed to version control)
 ├── notebooks       # Project-related Jupyter notebooks (can be used for experimental code before moving the code to src)
-├── pyproject.toml  # Identifies the project root and [contains configuration information](../faq/architecture_overview.md#kedro-project)
+├── pyproject.toml  # Identifies the project root and contains configuration information
```
See above about linking to project settings page

stichbury replied:
Same comment in return 😆  -- I'm not seeing the `.toml` project documentation at present, only a passing mention to it in the (now defunct) architecture page. When that is added, we should definitely add some links to it.

astrojuanlu commented on docs/source/get_started/kedro_concepts.md:
```diff
@@ -90,7 +90,7 @@ By default, Kedro creates one file, `credentials.yml`, in `conf/local`.
 ### `data`
 
-The `data` folder contains multiple subfolders to store project data. We recommend you put raw data into `raw` and move processed data to other subfolders according to the [data engineering convention](../faq/faq.md#what-is-data-engineering-convention).
+The `data` folder contains multiple subfolders to store project data. We recommend you put raw data into `raw` and move processed data to other subfolders according to the [data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71).
```
The sentence "The data engineering convention" has always bugged me, like there's only one. In any case, let's tackle this along with https://github.com/kedro-org/kedro-devrel/issues/42 ?

astrojuanlu commented on docs/source/kedro_project_setup/settings.md:
````diff
@@ -3,7 +3,7 @@
 
 ```{note}
-Application settings is distinct from [run time configuration](configuration.md), which is stored in the `conf` folder and can vary by configuration environment, and [pyproject.toml](../faq/architecture_overview.md#kedro-project) , which provides project metadata and build configuration.
+Application settings is distinct from [run time configuration](configuration.md), which is stored in the `conf` folder and can vary by configuration environment, and according to the contents of `pyproject.toml`, which provides project metadata such as `package_name`, `project_name` and `project_version` (the version of Kedro with which the project was generated).
````
See above

stichbury replied:
Ditto. I added this bit of text from the architecture doc as I removed it. It doesn't mention the `context_path` though, which is what you were referring to previously. I think I may need a bit of help with this (adding more docs about `pyproject.toml`)

astrojuanlu commented on docs/source/resources/glossary.md:
```diff
@@ -36,7 +36,7 @@ You can use Kedro-Viz to visualise your Kedro data pipelines:
 
 ## Layers (data engineering convention)
-According to [data engineering convention](../faq/faq.md#what-is-data-engineering-convention), a pipeline can be broken up into different layers according to how data is processed. This convention makes it easier to collaborate with other team members because everyone has an idea of what type of data cleaning or processing has happened.
+According to [data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71), a pipeline can be broken up into different layers according to how data is processed. This convention makes it easier to collaborate with other team members because everyone has an idea of what type of data cleaning or processing has happened.
```
See above

astrojuanlu replied:
Also, for this glossary, maybe we want to use Sphinx semantic tags? https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-term (not for this PR, and especially not if we end up deciding against Sphinx)

astrojuanlu commented on docs/source/tutorial/set_up_data.md:
```diff
@@ -6,7 +6,7 @@ Don't forget to check the [tutorial FAQ](spaceflights_tutorial_faqs.md) if you r
 In this section, we discuss the data setup phase, which is the second part of the standard development workflow. The steps are as follows:
 
-* Add datasets to your `data` folder, according to [data engineering convention](../faq/faq.md#what-is-data-engineering-convention)
+* Add datasets to your `data/raw` folder
```
👍🏽

astrojuanlu commented on docs/source/visualisation/kedro-viz_visualisation.md:
```diff
@@ -76,7 +76,7 @@ The `autoreload` flag reflects changes to the project as they happen. For exampl
 By convention, a [pipeline can be defined as having different layers](../resources/glossary.md#layers-data-engineering-convention) according to how data is processed, which makes it easier to collaborate.
 
-For example, the [data engineering convention](../faq/faq.md#what-is-data-engineering-convention) labels datasets according to the stage of the pipeline (e.g. whether the data has been cleaned).
+For example, the [data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71) labels datasets according to the stage of the pipeline (e.g. whether the data has been cleaned).
```
See above

astrojuanlu reviewed (commented):
Thanks a lot @stichbury! Left a few comments

stichbury commented:
I am not going to merge this PR until we have the following in place:

[* Agreement on general messaging](https://github.com/kedro-org/kedro-devrel/issues/2)
* Updated copy for "marketing" FAQs (these are in Box)
* Transitioned those FAQs from their location in Box [onto website](https://github.com/kedro-org/kedro-website/issues/71)
* Added a set of basic technical FAQs to the FAQ page in the docs

ankatiyar commented on docs/source/contribution/technical_steering_committee.md:
```diff
@@ -59,4 +62,6 @@ GitHub discussions is used to host votes on issues, proposals and changes affect
-Adding or removing maintainers requires a pull request against [our team list](../faq/faq.md#who-maintains-kedro). The decision is made based on TSC members votes in that pull request. Additions and removals of maintainers require **a 2/3 majority**.
+The decision to add or remove a maintainer is made based on TSC members votes in that pull request. Additions and removals of maintainers require **a 2/3 majority**.
+
+The act of adding or removing maintainers onto the list requires a pull request against [Kedro's README.md file](https://github.com/kedro-org/kedro/blob/main/README.md#the-humans-behind-kedro).
```
I think line 65 and 67 might be in the wrong order.

merelcht commented on docs/source/tutorial/tutorial_template.md:
```diff
@@ -87,12 +87,18 @@ To install all the project-specific dependencies, run the following from the pro
 To do this, add them to `conf/local/credentials.yml` (some examples are included in that file for illustration).
 
-You can find additional information in the [advanced documentation on configuration](../kedro_project_setup/configuration.md).
+### Configuration best practice to avoid leaking confidential data?
```
```suggestion
### Configuration best practice to avoid leaking confidential data
```

Or was this meant as a question?

merelcht reviewed (approved):
This looks really good! ⭐  I left one minor comment.
